modules/platforms/cpp/ignite/client/detail/node_connection.cpp

```diff
@@ -203,7 +208,46 @@ std::shared_ptr<response_handler> node_connection::find_handler_unsafe(std::int6
     if (it == m_request_handlers.end())
         return {};
 
-    return it->second;
+    return it->second.handler;
+}
+
+void node_connection::handle_timeouts() {
+    auto now = std::chrono::steady_clock::now();
+
+    {
+        std::lock_guard lock(m_request_handlers_mutex);
+
+        std::vector<int64_t> keys_for_erasure;
+
+        for (auto& [id, req] : m_request_handlers) {
+            if (req.timeouts_at > now) {
```

Comment:
It seems wrong. Do tests pass?
```suggestion
            if (req.timeouts_at < now) {
```